request.cpp

```diff
@@ -0,0 +1,23 @@
+#include "request.h"
+
+Request::Request(const std::string& start_str)
+    : m_verb(start_str.substr(0, 3)),
+      m_path(start_str.substr(4, start_str.length() - 4 - 9)),
+      m_version(start_str.substr(start_str.length() - 8))
```

Comment:
What id the version is not `HTTP/1.1` but `ICY`? The `m_version` will be incorrect.